app/controllers/data_access_controller.rb

```diff
@@ -192,6 +197,26 @@ def options
 
   private
 
+  # forms in methods are fresh, latest ones.
+  # The digest in access_config may be dirty.
+  def echo_form_outdated?(access_config, methods)
+    return true if access_config.nil? || access_config.echoform_digest.nil? || methods.nil?
+
+    form_digest = []
+    methods.each do |method|
+      if method[:type] == 'download'
+        digest = access_config.echoform_digest.select {|digest| digest['id'] == method[:type]}
+      else
+        return true if method[:form].nil?
```

Comment:
This isn't right, is it? If there is no form for an order (which I think is possible) it needs to be selected as default.

I think this method is potentially overcomplicated. Check if the form has changed (to include the addition / removal of a form). If it has, return false, otherwise true.